### ISSUE TITLE
Support UNIX sockets for db connections

### ISSUE THREAD

**Issue by rysiekpl** [sameersbn/docker-gitlab#415]
Reading [the documentation](https://hub.docker.com/r/sameersbn/gitlab/) I cannot find how to configure GitLab docker container to connect to a database (in my case: PostgreSQL) via a UNIX socket. If UNIX sockets are not supported, it would be worthwhile to add such a support, methinks!

**Comment by lazyfrosch** [comment]
The idea of Docker is to isolate applications, so they can only use networking to communicate...

**Comment by rysiekpl** [comment]
The idea of Docker is to isolate applications, full stop. Using UNIX sockets for applications in a single group (say, managed by `docker-compose`) makes it possible to not have certain containers/applications listening on *any* network ports, which adds some additional security.

Moreover, it makes it possible not to worry about "which IP address this container will get *this time*". If we're talking UNIX sockets through [data volume containers](http://docs.docker.com/userguide/dockervolumes/), we don't have to worry about network stuff where network stuff is not needed.

UNIX sockets can also be actually faster than TCP/IP stack in certain scenarios.

**Comment by sameersbn** [comment]
i agree that we need to add support for unix sockets. but due to time limitations this is not very high on my priority list as this will only have any lasting performance improvements on large installations me thinks.

**Comment by ivlis** [comment]
I have implemented a UNIX socket connection via a DVC from gitlab to ngnix. Making connections to databases is not a big deal. After @sameersbn will finish porting 8.0.0 I will create a PR.

**Comment by sameersbn** [comment]
p.s. the sameersbn/postgresql and sameersbn/mysql images already expose volumes that contain the unix domain sockets. Simply doing a `--volumes-from` would make this volume available to the gitlab container. Just a couple of changes are required in the `entrypoint.sh` script to detect and configure `database.yml` accordingly.

> After @sameersbn will finish porting 8.0.0 I will create a PR.

This is for the best.

**Comment by rysiekpl** [comment]
Any news on this?

**Comment by sameersbn** [comment]
@ivlis I will be happy to review your PR with this feature.

**Comment by ivlis** [comment]
@rysiekpl  @sameersbn Thanks for reminding me about that :smile:.  Gitlab is changing rapidly, I'll update my config to the current version and submit a PR soon.
